src/Microsoft.Sbom.Api/Utils/Comparer/SbomPackageComparer.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Sbom.Contracts;
+
+namespace Microsoft.Sbom.Api.Utils.Comparer;
+
+public class SbomPackageComparer : IEqualityComparer<SbomPackage>
+{
+    private static readonly SbomChecksumComparer ChecksumComparer = new SbomChecksumComparer();
+
+    public bool Equals(SbomPackage package1, SbomPackage package2)
+    {
+        if (package1 is null || package2 is null)
+        {
+            return false;
+        }
+
+        // Normalize PackageUrl for root packages since the SBOM tool version will vary between SPDX 2.2 and SPDX 3.0.
+        // Also normalize the Checksum since SPDX 2.2 does not support PackageVerificationCode to Checksum conversions for Root packages.
+        if (package1.Id == Constants.RootPackageIdValue && package2.Id == Constants.RootPackageIdValue)
+        {
+            package1.PackageUrl = NormalizePackagePurl(package1);
```

Comment:
I wouldn't expect an `Equals` method to modify the underlying data. Can we save the localized version in a local variable, instead?

Reply:
This is still changing the inputs